### PR TITLE
Release Godot 4.2.1 RC 1 and publish release notes

### PR DESCRIPTION
**Pull request by akien-mga**

### CODE REVIEW

File: _data/versions.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- name: "4.3"
+  flavor: "rc1"
+  release_date: "8 December 2023"
+  release_notes: "/article/release-candidate-godot-4-2-1-rc-1/"
```
Comment (akien-mga) [inline _data/versions.yml]: @YuriSizov Is it correct not to have a `releases` section when there's only one release?

Comment (akien-mga) [reply _data/versions.yml]: Of course it's better with `name: "4.2.1"` :P

Comment (YuriSizov) [reply _data/versions.yml]: Nothing should break if it's amiss, so yeah

File: collections/_article/release-candidate-godot-4-2-1-rc-1.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+-----
+
+*The illustration picture for this article comes from* [**Subzero Tides**](https://store.steampowered.com/app/2544930/Subzero_Tides/), *a sci-fi Moon colony builder developed by [Logical Progression Games](https://logicalprogression.tech/), recently upgraded to Godot 4.2. You can [wishlist the game on Steam](https://store.steampowered.com/app/2544930/Subzero_Tides/), check out their previous Godot-made action RPG [Terra Ventura](https://store.steampowered.com/app/1664580/Terra_Ventura/), and follow the developer on [Twitter](https://twitter.com/LogicalProgres3).*
+
+## What's new
+
+**41 contributors** submitted around **69 improvements** for this release. You can review the complete list of changes with our [interactive changelog](https://godotengine.github.io/godot-interactive-changelog/#4.2.1-rc1), which contains links to relevant commits and PRs for this and every previous release. Below are the most notable changes:
```
Comment (YuriSizov) [inline collections/_article/release-candidate-godot-4-2-1-rc-1.md:30]: Nice

**Review by YuriSizov — APPROVED** [review]
Looks good to me overall, and I don't have energy to nitpick beyond that 🙃

**Pull request merged**